pygmt/tests/test_text.py

```diff
@@ -96,6 +97,49 @@ def test_text_nonexistent_filename():
         fig.text(region=[10, 70, -5, 10], textfiles=[POINTS_DATA, "notexist.txt"])
 
 
+@pytest.mark.mpl_image_compare
+def test_text_position_four_corners(region):
+    """
+    Print text at four corners (top left/right, bottom left/right) of map.
+    """
+    fig = Figure()
+    for position in ("TL", "TR", "BL", "BR"):
+        fig.text(
+            region=region, projection="x1c", frame="a", position=position, text=position
```

Comment:
Two issues here:

1. Every time `frame='a'` is used, a basemap/frame is plotted. Here, the same basemap is plotted 4 times. It's not a big issue here, but if a basemap is plotted hundreds of times, the frame will look blurry. It's better to use the `basemap()` function to plot a basemap before calling `text()`.

2. The baseline image is incorrect. Here is what I'm getting using the equivalent command-line version:
```
gmt begin map png
echo TL | gmt text -Jx1c -R0/5/0/2.5 -Ba -F+cTL
echo TR | gmt text -Jx1c -R0/5/0/2.5 -Ba -F+cTR
echo BL | gmt text -Jx1c -R0/5/0/2.5 -Ba -F+cBL
echo BR | gmt text -Jx1c -R0/5/0/2.5 -Ba -F+cBR
gmt end show
```
Note the differences in **TL** and **BL**.
![image](https://user-images.githubusercontent.com/3974108/85190310-2dfa6300-b285-11ea-9ba5-4aa8120d2717.png)

Reply:
>     1. Every time `frame='a'` is used, a basemap/frame is plotted. Here, the same basemap is plotted 4 times. It's not a big issue here, but if a basemap is plotted hundreds of times, the frame will look blurry. It's better to use the `basemap()` function to plot a basemap before calling `text()`.

Good point. I would still rather have the unit tests use `text` only, so I'll add a centre-middle (CM) text to set up the plot and then loop through the four corners.

>     2. The baseline image is incorrect. Here is what I'm getting using the equivalent command-line version:
> Note the differences in **TL** and **BL**.

I did wonder why the left text were offset by a bit. I'll recreate/clean up my dev environment and try generating the images again.

Reply:
Ok, I've fixed point 1 (not plotting the frame 4 times), but still having trouble with point 2. I can reproduce your plot using the bash script, but somehow when using pygmt, 'TL' and 'BL' are still offset slightly. The thing is, I don't see that test failing in our CI either :confused: